state/factory/workingset.go

```diff
@@ -93,34 +93,6 @@ func (ws *workingSet) validate(ctx context.Context) error {
 	return nil
 }
 
-func (ws *workingSet) runActions(
-	ctx context.Context,
-	elps []action.SealedEnvelope,
-) ([]*action.Receipt, error) {
-	if err := ws.validate(ctx); err != nil {
-		return nil, err
-	}
-	// Handle actions
-	receipts := make([]*action.Receipt, 0)
-	for _, elp := range elps {
-		ctx, err := withActionCtx(ctx, elp)
```

Comment:
this is not clear, but are you sure it is a bug as well?

Reply:
we could fix by
```
actionCtx, err := withActionCtx(ctx, act)
```
and use `actionCtx`, this would prevent overwriting `ctx`